### PR TITLE
Fix SE-SE4 not being hoverable

### PR DESCRIPTION
**Pull request by tonypls**
## Issue
https://linear.app/electricitymaps/issue/ELE-1455/fix-issue-with-se-se4-where-selected-state-stays-forever
## Description
It appears there's a maplibre or mapbox bug where when you try and reference an id:0 on hover or click it does exist.

I have injected an empty zone and index 0 so when we map the geometries the dead zone isn't part of our map. 

There might be a smarter way to handle this, up for suggestions.

### CODE REVIEW

File: web/src/features/map/map-utils/generateTopos.ts [hunk 1]
```diff
@@ -31,7 +31,7 @@ export interface Topo {
 const generateTopos = (theme: MapTheme, spatialAggregate: string): MapGeometries => {
   const geometries: MapGeometries = { features: [], type: 'FeatureCollection' };
   const topography = topo as Topo;
-
+  geometries.features.push({ properties: { zoneId: 'emptyZone' } } as any);
```
Comment (madsnedergaard) [inline web/src/features/map/map-utils/generateTopos.ts:34]: Could you add a comment here describing why we're doing this? :)

**Review by madsnedergaard — APPROVED** [review]
Nice catch! And weird... Is it something on our end in the logic around hovering/clicking? It's weird that this wasn't a problem before :)
It would be nice to document why we're adding this hack, otherwise I'm fine with it

**Comment by tonypls** [comment]
> Nice catch! And weird... Is it something on our end in the logic around hovering/clicking? It's weird that this wasn't a problem before :) It would be nice to document why we're adding this hack, otherwise I'm fine with it

It appears to have been an issue here in mapbox that's been fixed - https://github.com/mapbox/mapbox-gl-js/issues/11994

It could be a bug that was brought over to maplibre and hasn't been fixed. I think the old app had a different map library which could account for it only showing up now.

**Comment by Kongkille** [comment]
I can confirm that the 0 index is super buggy to work with

Anyways, I looked into it a bit and I think a better solution is to use the zoneId as the uniqueId we work with in the map, this way we don't have to worry about 0 index and it's also nice to work with a relatable id.

Turns out this is possible using 'promoteId' in the source, this seems to work fine.

**Comment by tonypls** [comment]
> I can confirm that the 0 index is super buggy to work with
> 
> Anyways, I looked into it a bit and I think a better solution is to use the zoneId as the uniqueId we work with in the map, this way we don't have to worry about 0 index and it's also nice to work with a relatable id.
> 
> Turns out this is possible using 'promoteId' in the source, this seems to work fine.
> 
> I'll let you merge it @tonypls if you agree with this solution

Nice, yes i tried using the zoneId as it seemed to make sense but I didn't know about this magic promote id. ty

**Pull request merged**